src/skin/legacyskinparser.cpp

```diff
@@ -1084,6 +1089,30 @@ QWidget* LegacySkinParser::parseCoverArt(QDomElement node) {
     return pCoverArt;
 }
 
+void LegacySkinParser::parseSingletonDefinition(QDomElement node) {
+    // The actual singleton definition is identical to any other WWidgetGroup.
+    QWidget* child_group = parseWidgetGroup(node);
+    m_pContext->getSingletonMap()->defineSingleton(child_group->objectName(),
+                                                   child_group);
+    child_group->hide();
+}
+
+QWidget* LegacySkinParser::parseSingletonContainer(QDomElement node) {
```

Comment:
> The idea is to keep the interface too simple to mess up -- tell the map "here is a new singleton," and later "give me the instance of the singleton with this name.

(sorry for continuing to hammer on such a minor thing)

It's "here is a new singleton," and later "give me the instance of the singleton with this name and also create a SingletonContainer without setting it up in the way widgets are normally set up and create it outside of LegacySkinParser which was previously the one-stop-shop to see where widgets are created." 

I understand wanting it to be simple but I'd say having more widgets use parseStandardWidget is simpler for the skin system overall because you don't have to read the code to know which widget types get event filters bound, which widget types get their Init method called (WSingletonContainer doesn't currently), which widget types get setup called, and in what order all of those methods gets called. This has caused headaches in the past (every widget parse method did things in its own order or in its own non-standard way) and I combined as many as possible into parseStandardWidget when I was pretty sure the non-standardness was accidental and not important. 

Ok I'm done -- feel free to keep it or change it. If you keep it the way it is please copy parseStandardWidget and add a note to keep it in sync with parseStandardWidget. :)